### PR TITLE
Add docs specifically for quo2

### PR DESCRIPTION
**Pull request by ilmotta**
## Summary

Here's a quick PR finally documenting our guidelines for the quo2 library.

Click on these links to read them rendered by GitHub:

- https://github.com/status-im/status-mobile/tree/update-docs-and-document-quo-lib/src/quo2
- https://github.com/status-im/status-mobile/blob/update-docs-and-document-quo-lib/doc/new-guidelines.md

### Notes

The guidelines for quo2 are rather succinct, so I didn't use the collapsible sections with the `details` HTML tag.

status: ready

### CODE REVIEW

File: src/quo2/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+quo2/
+└── components/
+    └── banners/
+        └── banner/
+            ├── component_spec.cljs
+            ├── style.cljs
+            └── view.cljs
+```
+
+Files `style.cljs` and `component_spec.cljs` should always have the same name,
```
Comment (cammellos) [inline src/quo2/README.md]: What about view?

**Review by cammellos — APPROVED** [review]

File: src/quo2/README.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  (:require [quo2.theme :as quo.theme]))
+
+(defn- view-internal [{:keys [theme]}]
+  ...)
+
+(def view (quo.theme/with-theme view-internal))
+```
+
+Then pass the `theme` value down to all functions that may rely on the OS theme,
+like `quo2.foundations.colors/custom-color-by-theme` or
```
Comment (J-Son89) [inline src/quo2/README.md]: let's remove this `quo2.foundations.colors/custom-color-by-theme` from here. This function is not setup to handle theme correctly and in fact I think we can move away from it. 

Let's just use `quo2.foundations.colors/theme-colors`

File: src/quo2/README.md (hunk 1 above)
Comment (ilmotta) [reply src/quo2/README.md]: Missed that, thanks.

File: src/quo2/README.md (hunk 2 above)
Comment (ilmotta) [reply src/quo2/README.md]: In the text I just wanted to mention one or two functions to make the point clearer. I'll use `theme-colors` in the example.

Comment (ilmotta) [reply src/quo2/README.md]: Done

File: src/quo2/README.md (hunk 1 above)
Comment (ilmotta) [reply src/quo2/README.md]: Done

File: src/quo2/README.md (hunk 2 above)
Comment (J-Son89) [reply src/quo2/README.md]: Thanks!

**Review by ajayesivan — APPROVED** [review]
Fantastic job on documenting this! 💪

**Review by siddarthkay — APPROVED** [review]
the quo2 readme is very good!

**Review by qfrank — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]
image

**Review by jo-mut — APPROVED** [review]

**Review by ulisesmac — APPROVED** [review]
Nice!!! :100:

**Review by OmarBasem — APPROVED** [review]
Great job @ilmotta ⭐

**Review by Parveshdhull — APPROVED** [review]

**Review by alwx — APPROVED** [review]

**Pull request merged**